src/common/perf_counters.cc

```diff
@@ -354,9 +380,26 @@ void PerfCounters::reset()
 }
 
 void PerfCounters::dump_formatted_generic(Formatter *f, bool schema,
-    bool histograms, const std::string &counter) const
+    bool histograms, bool dump_labeled, const std::string &counter) const
 {
-  f->open_object_section(m_name.c_str());
+  if(dump_labeled) {
+    std::string_view perf_counter_name = ceph::perf_counters::key_name(m_name);
+    f->open_object_section(perf_counter_name);
+
+    f->open_object_section("labels");
+    for (auto label : ceph::perf_counters::key_labels(m_name)) {
+      // don't dump labels with empty label names or values
+      if(label.first == "" || label.second == "") {
+        continue;
```

Comment:
what's wrong with empty keys/values?

Reply:
We discussed that in a meeting, and we commented that labels without a label name are unusable labels.

Reply:
As a group we agreed that a counters that could somehow be: `"rgw": {
        "labels": {
            "Bucket": "",
            "User": "user1",
        }
        "put_b": 1048576,
    }` or `"rgw": {
        "labels": {
            "": "bkt2",
            "User": "user1",
        }
        "put_b": 1048576,
    }` appear as `"rgw": {
        "labels": {
            "User": "user1",
        }
        "put_b": 1048576,
    }`

Reply:
i think it's important to support empty values. taking the "User" label as an example, rgw supports unauthenticated requests. `{'User': ''}` seems like the natural way to label those requests. if you filter those out, users wouldn't have a way to view the counters for unauthenticated requests